todo_app/data/trello_items.py

```diff
@@ -0,0 +1,146 @@
+        raise ValueError(f'Could not find configured to do board: {configured_todo_board_name}')
+
+    return todo_board['id']
+
+def get_lists() -> List[dict]:
+    todo_board_id = get_todo_board_id()
+    return requests.get(f'{base_uri}/boards/{todo_board_id}/lists', params=query_params).json()
+
+def get_list_id_or_throw(lists: List[dict], list_name: str) -> str:
+    list = next(iter([x for x in lists if x['name'] == list_name]), None)
+    if (list is None):
+        raise ValueError(f'Could not find list "{list_name}". Make sure you have this configured on your trello board')
+    return list['id']
+
+def get_items_from_list(list_id: str, status: str) -> List[Item]:
+    cards = requests.get(f'{base_uri}/lists/{list_id}/cards', params=query_params).json()
+    return [Item(id=card['id'], title=card['name'], status=status) for card in cards]
+
+def create_todo_item(title: str):
+    lists = get_lists()
+    to_do_list_id = get_list_id_or_throw(lists, "To Do")
+    response = requests.post(f'{base_uri}/cards', params={**query_params, 'name': title, 'idList': to_do_list_id})
+
+    if response.status_code != 200:
+        raise ValueError(f'Failed to create todo item for "{title}". Received status code {response.status_code}')
+    
+    response_body = response.json()
+    return Item(id=response_body['id'], title=title, status=ItemStatus.NOT_STARTED)
+
+def remove_item(id: str):
+    response = requests.delete(f'{base_uri}/cards/{id}', params=query_params)
+    if response.status_code != 200:
+        raise ValueError(f'Failed to delete todo item with id "{id}"". Received status code {response.status_code}')
+
+def update_item(item: Item):
+    lists = get_lists()
+    to_do_list_id = get_list_id_or_throw(lists, "To Do")
+    doing_list_id = get_list_id_or_throw(lists, "Doing")
+    done_list_id = get_list_id_or_throw(lists, "Done")
+    
+    list_id_to_move_to = None
+    if item.status == ItemStatus.NOT_STARTED: list_id_to_move_to = to_do_list_id
+    elif item.status == ItemStatus.IN_PROGRESS: list_id_to_move_to = doing_list_id
+    elif item.status == ItemStatus.COMPLETE: list_id_to_move_to = done_list_id
+
+    response = requests.put(f'{base_uri}/cards/{item.id}', params={**query_params, 'idList': list_id_to_move_to})#
+
+    if response.status_code != 200:
+        raise ValueError(f'Failed to update todo item with id "{item.id}". Received status code {response.status_code}')
+
+
+def get_items() -> List[Item]:
+    """
+    Fetches all saved items from trello.
+
+    Returns:
+        list: The list of saved items.
+    """
+    lists = get_lists()
+    to_do_list_id = get_list_id_or_throw(lists, "To Do")
+    doing_list_id = get_list_id_or_throw(lists, "Doing")
+    done_list_id = get_list_id_or_throw(lists, "Done")
+
+    to_do_items = get_items_from_list(to_do_list_id, 'Not Started')
+    in_progress_items = get_items_from_list(doing_list_id, 'In Progress')
+    complete_items = get_items_from_list(done_list_id, 'Complete')
```

Comment:
Similarly, if you found that the number of Trello calls you're making is slowing your application down, switching this logic to instead [get all of the cards on the board](https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-cards-get) might prove more efficient